src/network/interfacemanager.cpp

```diff
@@ -0,0 +1,511 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <algorithm>
+#include <memory>
+#include <string>
+
+#include <arpa/inet.h>
+#include <linux/if.h>
+#include <linux/rtnetlink.h>
+#include <netinet/in.h>
+#include <netlink/netlink.h>
+#include <netlink/route/addr.h>
+#include <netlink/route/link.h>
+#include <netlink/route/link/bridge.h>
+#include <netlink/route/link/vlan.h>
+#include <netlink/route/route.h>
+
+#include "logger/logmodule.hpp"
+#include "network/interfacemanager.hpp"
+
+namespace aos::common::network {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+Bridge::Bridge(const LinkAttrs& attrs)
+    : mAttrs(attrs)
+{
+}
+
+const LinkAttrs& Bridge::GetAttrs() const
+{
+    return mAttrs;
+}
+
+const char* Bridge::GetType() const
+{
+    return "bridge";
+}
+
+Vlan::Vlan(const LinkAttrs& attrs, int vlanId)
+    : mAttrs(attrs)
+    , mVlanId(vlanId)
+{
+}
+
+const LinkAttrs& Vlan::GetAttrs() const
+{
+    return mAttrs;
+}
+
+const char* Vlan::GetType() const
+{
+    return "vlan";
+}
+
+int Vlan::GetVlanId() const
+{
+    return mVlanId;
+}
+
+Error InterfaceManager::DeleteLink(const String& ifname)
+{
+    LOG_DBG() << "Remove interface: ifname=" << ifname;
+
+    auto [sock, err] = CreateNetlinkSocket();
+    if (!err.IsNone()) {
+        return err;
+    }
+
+    auto [link, linkErr] = CreateLink();
+    if (!linkErr.IsNone()) {
+        return linkErr;
+    }
+
+    rtnl_link_set_name(link.get(), ifname.CStr());
+
+    if (auto errLinkDel = rtnl_link_delete(sock.get(), link.get()); errLinkDel < 0) {
+        return Error(ErrorEnum::eFailed, ("failed to delete link: " + std::string(nl_geterror(errLinkDel))).c_str());
```

Comment:
consider to implement `NLToAorErr` function.

Reply:
Done